server/controllers/authController.ts

```diff
@@ -56,3 +60,71 @@ export const signIn = async (req: Request, res: Response) => {
     res.status(404).json({ message: error.message });
   }
 };
+
+// *** NEW AUTH ***
+
+export const register = async (req, res) => {
+  const { name, email, password } = req.body;
+
+  try {
+    const hashedPassword = await hash(password, 10);
+    await prisma.user.create({
+      data: { name, email, password: hashedPassword },
+    });
+
+    return res.status(201).json({
+      success: true,
+      message: "The registration was successful",
+    });
+  } catch (error) {
+    console.log(error);
+    return res.status(500).json({
+      error: error.message,
+    });
+  }
+};
+
+export const login = async (req, res) => {
+  // This was passed down from the loginFieldsCheck function in Validators/authVal
+  let user = req.user;
+
+  const payload = {
+    id: user.id,
+    email: user.email,
+    name: user.name,
+    createdAt: user.createdAt,
+  };
+
+  try {
+    // Get JWT token
+    const token = await sign(payload, constants.SECRET);
+    console.log("SIGNIN RAN", token);
+
+    // Pass cookie
+    return res.status(200).cookie("token", token, { httpOnly: true }).json({
+      success: true,
+      message: "Logged in successfully",
+      token: token,
+    });
+  } catch (error) {
+    console.log(error.message);
+    return res.status(500).json({
+      error: error.message,
+    });
+  }
+};
+
+export const userData = async (req, res) => {
```

Comment:
why we need a api endpoint in authController that just returns a user??
-not sure but I dont think we should have the user-email in the url parameters

Reply:
To get user data in the frontend